### PR TITLE
[parachain-template] pallet configurations into `mod configs`

### PR DESCRIPTION
**Pull request by asiniscalchi**
This PR introduces a refactor of the parachain runtime configuration by consolidating all pallet configurations into a new module named `configs`. This change aims to improve the readability and maintainability of the runtime configuration by centralizing all configuration parameters.

## Changes
- **Creation of `configs.rs`**: A new file `configs.rs` has been added under `templates/parachain/runtime/src/`, containing all the runtime configurations previously scattered across `lib.rs`.
- **Refactoring of `lib.rs`**: The `lib.rs` file has been significantly slimmed down by removing the inline pallet configurations and importing them from `configs.rs` instead.
- **Optimization of Import Statements**: Reorganized import statements to clarify the runtime's dependency structure. 

### Benefits
- **Improved Readability**: With configurations being centralized, developers can now easily locate and review runtime parameters without navigating through the `lib.rs` file.

This refactor does not introduce any changes to the runtime logic but improves the project structure for better development experience.

### CODE REVIEW

**Review by bkchr — APPROVED** [review]

**Comment by asiniscalchi** [comment]
Hi, @kianenigma sorry to `ping`

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5685637

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5685788

**Review by ggwpez — APPROVED** [review]

**Comment by asiniscalchi** [comment]
@kianenigma ping :)

File: templates/parachain/runtime/src/configs/mod.rs [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+		.avg_block_initialization(AVERAGE_ON_INITIALIZE_RATIO)
+		.build_or_panic();
+	pub const SS58Prefix: u16 = 42;
+}
+
+/// The default types are being injected by [`derive_impl`](`frame_support::derive_impl`) from
+/// [`ParaChainDefaultConfig`](`struct@frame_system::config_preludes::ParaChainDefaultConfig`),
+/// but overridden as needed.
+#[derive_impl(frame_system::config_preludes::ParaChainDefaultConfig)]
+impl frame_system::Config for Runtime {
```
Comment (kianenigma) [inline templates/parachain/runtime/src/configs/mod.rs:99]: Are all the ones that are mentioned here actually overwriting something? `type Hash`, `type BlockWeight` etc should be the same in `ParachainDefaultConfig`

**Review by kianenigma — APPROVED** [review]
This is a good step, but if interested, I would suggest you to: 

- provide sensible defaults for `Solochain` and `Parachain` for: `balances`, `system`, `timestamp`, `transaction-payment`, `sudo`. 
- Update all 3 templates to use these defaults. 
- @gupnik can provide supervision.

**Comment by asiniscalchi** [comment]
> This is a good step, but if interested, I would suggest you to:
> 
> * provide sensible defaults for `Solochain` and `Parachain` for: `balances`, `system`, `timestamp`, `transaction-payment`, `sudo`.
> * Update all 3 templates to use these defaults.
> * @gupnik can provide supervision.

Here you are: https://github.com/paritytech/polkadot-sdk/issues/3916

**Pull request merged**